motor-control/firmware/motor_control_hardware.c

```diff
@@ -94,15 +94,35 @@ bool motor_hardware_stop_pwm(void* htim, uint32_t channel) {
  * read a hal timer count value
  * if the timer is NULL return 0
  */
-uint16_t _get_hal_timer_count(void* htim) {
+uint32_t _get_hal_timer_count(void* htim) {
     if (htim != NULL) {
         return __HAL_TIM_GET_COUNTER((TIM_HandleTypeDef*)htim);
     }
     return 0;
 }
 
-int32_t motor_hardware_encoder_pulse_count(void* enc_htim) {
-    int32_t pulses = _get_hal_timer_count(enc_htim);
+uint32_t motor_hardware_encoder_pulse_count(void* enc_htim) {
+    return _get_hal_timer_count(enc_htim) & 0xFFFF;
+}
+
+bool motor_hardware_encoder_is_counting_down(void *encoder_handle) {
+    if(encoder_handle == NULL) {
+        return true;
+    }
+    return __HAL_TIM_IS_TIM_COUNTING_DOWN((TIM_HandleTypeDef*)encoder_handle);
+}
+
+uint32_t motor_hardware_encoder_pulse_count_with_overflow(void* encoder_handle, int8_t *overflows) {
```

Comment:
```suggestion
uint16_t motor_hardware_encoder_pulse_count_with_overflow(void* encoder_handle, int8_t *overflows) {
```
Let's make this the same return type as the other one just for cleanliness

Reply:
af28af4da7c0ae0df97e4aef333c752f79f4abf4